### PR TITLE
fix(typescript): compatible with 4.3

### PR DESCRIPTION
**Pull request by smasala**
Fixes (see: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-3.html): 
```
TS4114: This member must have an 'override' modifier because it overrides a member in the base class 'Error'.
    65 |  */
    66 | export class RequiredError extends Error {
  > 67 |     name: "RequiredError" = "RequiredError";
       |     ^^^^
    68 |     constructor(public field: string, msg?: string) {
    69 |         super(msg);
    70 |     }
```

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (5.3.0), `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by smasala** [comment]
CR Please: @TiFu @taxpon @sebastianhaas @kenisteward @Vrolijkx @macjohnny @topce @akehir @petejohansonxo @amakhrov @wing328

File: docs/generators/jaxrs-cxf-client.md [hunk 1]
```diff
@@ -49,7 +49,7 @@ These options may be applied as additional-properties (cli) or configOptions (pl
 |snapshotVersion|Uses a SNAPSHOT version.|<dl><dt>**true**</dt><dd>Use a SnapShot Version</dd><dt>**false**</dt><dd>Use a Release Version</dd></dl>|null|
 |sortModelPropertiesByRequiredFlag|Sort model properties to place required parameters before optional parameters.| |true|
 |sortParamsByRequiredFlag|Sort method arguments to place required parameters before optional parameters.| |true|
-|sourceFolder|source folder for generated code| |src/gen/java|
+|sourceFolder|source folder for generated code| |src\gen\java|
```
Comment (macjohnny) [inline docs/generators/jaxrs-cxf-client.md:52]: Please revert these path changes, probably due to generation on a windows system

File: modules/openapi-generator/src/main/resources/typescript-fetch/runtime.mustache [hunk 2]
```diff
@@ -102,7 +102,7 @@ export class BaseAPI {
 };
 
 export class RequiredError extends Error {
-    name: "RequiredError" = "RequiredError";
+    override name: "RequiredError" = "RequiredError";
```
Comment (macjohnny) [inline modules/openapi-generator/src/main/resources/typescript-fetch/runtime.mustache:105]: This change requires TS 4.3+, but the generated code should have compatibility with 2.4/3.9:

https://github.com/OpenAPITools/openapi-generator/blob/6392cc1876f54cc5bae1236586ae7f5630059bb7/modules/openapi-generator/src/main/resources/typescript-fetch/package.mustache#L26

**Comment by smasala** [comment]
@macjohnny closed in favour of https://github.com/OpenAPITools/openapi-generator/pull/11305

**Pull request closed without merging**